### ISSUE TITLE
[FEATURE REQUEST]: Documentation support

### ISSUE THREAD

**Issue by Niharikadutta** [dotnet/spark#671]
Documentation needs to be added about the following:

**New docs**:

1. How to's:
   i. Connect to ADLS gen2 account locally (#337)
   ii. Use .NET interactive notebooks locally (#529)
   iii. Invoke C# UDFs from Scala/Python - (import [this](https://github.com/imback82/sparkdotnetudf/blob/master/README.md#invoke-c-udf-from-python))

2. Dotnet-interactive guide/good to know's:
   i. Dotnet interactive serialization issue (#619)
   ii. Running UDFs in notebooks

**Update existing docs**:

3. Update [this](https://docs.microsoft.com/en-us/azure/synapse-analytics/spark/spark-dotnet) documentation to reflect how to zip binaries for .NET for Spark batch submission.

**Comment by rapoth** [comment]
@luisquintanilla @mrys @JeremyLikness Do you have any other topics we should cover (based on what you heard from users)?

**Comment by Niharikadutta** [comment]
For "Using .NET for Spark on Azure Synapse Analytics - Submitting batch jobs", should we just update [this](https://docs.microsoft.com/en-us/azure/synapse-analytics/spark/spark-dotnet) documentation instead of creating a new one?

**Comment by rapoth** [comment]
Yes, please! That's a great idea! Can you also update your issue with any existing links to docs that already exist and that need to be updated?

**Comment by luisquintanilla** [comment]
Not sure exactly where this would fit. Maybe it can be a standalone article, but basically, how can you make use of Scala libraries that run on top of Apache Spark within .NET for Apache Spark?

Some Examples: 
- [MMLSpark](https://github.com/Azure/mmlspark).
- [TransmogrifAI](https://github.com/salesforce/TransmogrifAI)

At the moment there are no .NET bindings for these libraries. Can I still use them with .NET for Spark? If so, how?

**Comment by luisquintanilla** [comment]
@Niharikadutta 

Just wanted to clarify what is meant for this work item. Does this mean, using .NET Interactive with .NET for Apache Spark?

Use .NET for Apache Spark locally using VS Code - @luisquintanilla working on this

**Comment by imback82** [comment]
Linking #293 as migration guide

**Comment by imback82** [comment]
Can we also capture the followings in the descrption?

1. Known issues: e.g.) grouped map UDF for Spark 3.0

2. .NET Delta/Hyperspace version | OSS Delta/Hyperspace version | Spark versions matrix

**Comment by imback82** [comment]
Can we also capture dropping support for .NET Framework?

**Comment by Niharikadutta** [comment]
@luisquintanilla As discussed offline, what `Use .NET for Apache Spark locally using VS Code` this work item means is using using .NET Interactive in VS Code with .NET for Apache Spark using the [.NET Interactive Notebooks extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-interactive-vscode).
And regarding your other question about using external Scala libraries through .NET for Apache Spark, I am working on that and will update the thread as soon as I have something concrete. Thanks!

**Comment by rapoth** [comment]
@indy-3rdman Do you have any feedback on this thread (specifically, if any additional topics need to be covered OR feedback on existing PRs which are linked to from the parent thread)? Thank you!

**Comment by rapoth** [comment]
@Niharikadutta Do you think we can close this issue?

**Comment by Niharikadutta** [comment]
> @Niharikadutta Do you think we can close this issue?

Yes we can close this.
